### PR TITLE
Measure Dashboard and Patient List Spreadsheet Export

### PR DESCRIPTION
**Pull request by smc-ssiddiqui**
- User can download spreadsheet of their measure dashboard in addition to QRDA Cat 3
- User can download spreadsheet of patients relative to a measure, i.e. outliers, numerators, denominators, etc.

### CODE REVIEW

File: app/controllers/api/reports_controller.rb [hunk 1]
```diff
@@ -62,6 +188,29 @@ def cat1
...
+        "Outlier"
+      when "DENEX"
+        "Exclusion"
+      else 
+        "N/A"
+      end
+    end
+
+    def percentage(numer, denom)	
+      percent = (numer/denom * 100).round(1)
```
Comment (eedrummer) [inline app/controllers/api/reports_controller.rb]: Will this line raise an exception if the denom == 0?

File: app/controllers/api/reports_controller.rb [hunk 2]
```diff
@@ -40,6 +40,132 @@ def cat3
...
+
+      today = Time.now.strftime("%D")  
+      filename = "patients_" + measure.cms_id + "_" + patient_type(type) + "_" + "#{today}" + ".xls"
+      data = StringIO.new '';
+      book.write data;
+      send_data(data.string, {
+        :disposition => 'attachment',
+        :encoding => 'utf8',
+        :stream => false,
+        :type => 'application/excel',
```
Comment (eedrummer) [inline app/controllers/api/reports_controller.rb]: Consider changing to "application/vnd.ms-excel", see: http://stackoverflow.com/questions/974079/setting-mime-type-for-excel-document

File: app/controllers/api/reports_controller.rb [hunk 3]
```diff
@@ -40,6 +40,132 @@ def cat3
...
+        selected_measures = measure_ids.map{ |id| HealthDataStandards::CQM::Measure.where(:id => id)}
+        # report header
+        provider = Provider.find(params[:provider_id])
+        authorize! :read, provider
+
+        eff = Time.at(params[:effective_date].to_i)
+        end_date = eff.strftime("%D")
+        start_date = eff.month.to_s + "/" + eff.day.to_s + "/" + (eff.year-1).to_s
+
+        sheet.row(0).push "Reporting Period: ", start_date + " - " + end_date
```
Comment (eedrummer) [inline app/controllers/api/reports_controller.rb]: Minor style nit: please use parenthesis. They seem to be omitted when calling the push method, but used everywhere else.

File: app/controllers/api/reports_controller.rb [hunk 4]
```diff
@@ -40,6 +40,132 @@ def cat3
...
+        sheet.row(1).push "Provider: ", provider.full_name
+        sheet.row(2).push "NPI: ", provider.npi
+        # table headers
+        sheet.row(4).push 'NQF ID', 'CMS ID', 'Sub ID', 'Title', 'Subtitle', 'Numerator', 'Denominator', 'Exclusions', 'Percentage'
+        sheet.row(4).default_format = format
+        (0..2).each do |i|
+          sheet.row(i).set_format(0, format)
+        end
+        r = 5
```
Comment (eedrummer) [inline app/controllers/api/reports_controller.rb]: This is a magic number (http://en.wikipedia.org/wiki/Magic_number_%28programming%29). Maybe you could set r = 0 on line 130 and increment it as you go along or create a constant to signify that the end of the header is at 5.

**Comment by eedrummer** [comment]
Overall, this looks good. I've made some minor comments directly on the code.

**Comment by petercyli** [comment]
The "pull request edits" looks good per Andy's suggestion.  I also re-performed the manual test.  Looks good for commit.

**Pull request merged**